python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -163,8 +163,39 @@ def router_multicast(self) -> None:
 
     @structured_config_contributor
     def hardware_counters(self) -> None:
-        """hardware_counters set based on hardware_counters.features variable."""
-        self.structured_config.hardware_counters = self.inputs.hardware_counters
+        """
+        Set hardware_counters.
+
+        Contributing data sources:
+          - hardware_counters.features variable.
+          - platform_settings.feature_support.hardware_counters fact.
+          - platform_settings.feature_support.hardware_counters_features fact.
+        """
+        if not self.inputs.hardware_counters:
+            return
+        # Avoid collision with eos_cli_config_gen
+        if not self.shared_utils.platform_settings.feature_support.hardware_counters:
+            self.custom_structured_configs.nested.hardware_counters = EosCliConfigGen.HardwareCounters._from_null()
+            return
+        hardware_counters = self.inputs.hardware_counters
```

Comment:
Since we may be in-place updating.
```suggestion
        hardware_counters = self.inputs.hardware_counters._deepcopy()
```

Reply:
Fixed